modules/synacormediaBidAdapter.js

```diff
@@ -79,41 +100,43 @@ export const spec = {
     }
   },
   interpretResponse: function(serverResponse) {
+    var updateMacros = (bid, r) => {
+      return r ? r.replace(/\${AUCTION_PRICE}/g, parseFloat(bid.price)) : r;
+    };
+
     if (!serverResponse.body || typeof serverResponse.body != 'object') {
       logWarn('Synacormedia: server returned empty/non-json response: ' + JSON.stringify(serverResponse.body));
       return;
     }
+
     const {id, seatbid: seatbids} = serverResponse.body;
     let bids = [];
     if (id && seatbids) {
       seatbids.forEach(seatbid => {
         seatbid.bid.forEach(bid => {
-          let price = parseFloat(bid.price);
-          let creative = bid.adm.replace(/\${([^}]*)}/g, (match, key) => {
-            switch (key) {
-              case 'AUCTION_SEAT_ID': return seatbid.seat;
-              case 'AUCTION_ID': return id;
-              case 'AUCTION_BID_ID': return bid.id;
-              case 'AUCTION_IMP_ID': return bid.impid;
-              case 'AUCTION_PRICE': return price;
-              case 'AUCTION_CURRENCY': return 'USD';
-            }
-            return match;
-          });
-          let [, impid, width, height] = bid.impid.match(/^(.*)~(.*)x(.*)$/);
-          bids.push({
+          let creative = updateMacros(bid, bid.adm);
+          let nurl = updateMacros(bid, bid.nurl);
+          let [, impType, impid, width, height] = bid.impid.match(/^([vb])(.*)-(.*)x(.*)$/);
+          let isVideo = impType == 'v';
+          let bidObj = {
             requestId: impid,
             adId: bid.id.replace(/~/g, '-'),
-            cpm: price,
+            cpm: parseFloat(bid.price),
             width: parseInt(width, 10),
             height: parseInt(height, 10),
-            creativeId: seatbid.seat + '~' + bid.crid,
+            creativeId: seatbid.seat + '_' + bid.crid,
             currency: 'USD',
             netRevenue: true,
-            mediaType: BANNER,
+            mediaType: (isVideo ? VIDEO : BANNER),
```

Comment:
Just a recommendation, but you could simplify by removing the parens here:
`mediaType: isVideo ? VIDEO : BANNER,`